### ISSUE TITLE
Failed To Update Config To Component Deployed odo push

### ISSUE THREAD

**Issue by elizabethlumban** [openshift/odo#4646]
/kind bug

## What versions of software are you using?

**Operating System:**

**Output of `odo version`:** 
odo v1.2.0
oc version:
Server Version: 4.5.35

## How did you run odo exactly?
I followed the instructions from this website

https://docs.openshift.com/container-platform/4.7/cli_reference/developer_cli_odo/creating_and_deploying_applications_with_odo/creating-a-multicomponent-application-with-odo.html

to deployed the following:
https://github.com/openshift-evangelists/Wild-West-Frontend
https://github.com/openshift-evangelists/Wild-West-Backend

I was trying to deploy this sample applications

## Actual behavior
When I execute the odo push I get a failend and this error message

Validation
 ✓  Checking component [216ms]
Configuration changes
 ✓  Retrieving component data [422ms]
 ✗  Applying configuration [5m]
 ✗  Failed To Update Config To Component Deployed.
Error: unable to update DeploymentConfig for backend component: unable to update the current DeploymentConfig backend-app: unable to wait for DeploymentConfig backend-app to update: Timed out waiting for annotation to update

## Any logs, error output, etc?

Validation
 ✓  Checking component [216ms]
Configuration changes
 ✓  Retrieving component data [422ms]
 ✗  Applying configuration [5m]
 ✗  Failed To Update Config To Component Deployed.
Error: unable to update DeploymentConfig for backend component: unable to update the current DeploymentConfig backend-app: unable to wait for DeploymentConfig backend-app to update: Timed out waiting for annotation to update

Thanks for your help

**Comment by elizabethlumban** [comment]
I updated my odo to this version odo v2.1.0 (8479050f6) I still got the same error :( 

Thanks in advance for your help

**Comment by dharmit** [comment]
@elizabethlumban we recently made a bunch of changes to the `odo create --s2i` command in #4518. Can you please update to use the latest master branch of odo and see if it helps?

**Comment by elizabethlumban** [comment]
hi @dharmit  would you be able to provide the steps on how to use the latest master branch of odo?

Thank you for the quick reply

**Comment by dharmit** [comment]
You'll need to clone the code to your system and build the binary.
```sh
$ git clone https://github.com/openshift/odo/
$ cd odo
$ make
```
Doing this will generate `odo` binary in your current working directory. You can then copy it to a location that's in your system's `$PATH` and use the odo binary.

**Comment by elizabethlumban** [comment]
hi @dharmit - I've followed the following steps but unfortunately, I still got the same error message.

**Comment by elizabethlumban** [comment]
hi, I just tried to perform odo push again today and I would like to share the following logs. I hope this would help. thanks

[odopush_26Apr2021.txt](https://github.com/openshift/odo/files/6383685/odopush_26Apr2021.txt)

**Comment by dharmit** [comment]
@elizabethlumban I just tried to follow the steps mentioned in the doc you've linked. And `odo push` worked for me for both the backend and frontend component. I need to look closely at the logs you've shared but did you add execute below steps before creating the components:
```sh
$ oc import-image openjdk18 \
--from=registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift --confirm

$ oc annotate istag/openjdk18:latest tags=builder
```

**Comment by dharmit** [comment]
Error message in the logs shared in https://github.com/openshift/odo/issues/4646#issuecomment-827483476:
```
I0426 17:35:45.140578   22726 pods.go:63] Status of java-maven-5fb6944575-mqj8f pod is Pending
I0426 17:35:45.141130   22726 pods.go:67] Pod Conditions: {"type":"PodScheduled","status":"False","lastProbeTime":null,"lastTransitionTime":"2021-04-23T09:31:46Z","reason":"Unschedulable","message":"running \"VolumeBinding\" filter plugin for pod \"java-maven-5fb6944575-mqj8f\": pod has unbound immediate PersistentVolumeClaims"}
I0426 17:35:45.386858   22726 round_trippers.go:443] GET https://ef4ea4857403842acc7b2-6b64a6ccc9c596bf59a86625d8fa2202-ce00.us-east.satellite.appdomain.cloud:30260/api/v1/namespaces/eol/events?watch=true 200 OK in 251 milliseconds
I0426 17:35:45.386918   22726 round_trippers.go:449] Response Headers:
I0426 17:35:45.386933   22726 round_trippers.go:452]     Cache-Control: no-cache, private
I0426 17:35:45.386944   22726 round_trippers.go:452]     Content-Type: application/json
I0426 17:35:45.386954   22726 round_trippers.go:452]     Date: Mon, 26 Apr 2021 09:35:45 GMT
I0426 17:39:45.144242   22726 events.go:36] Quitting collect events
 ✗  Waiting for component to start [4m]
 ✗  Failed to start component with name java-maven. Error: Failed to create the component: unable to get pod for component java-maven: error while waiting for pod component=java-maven: waited 4m0s but couldn't find running pod matching selector: 'component=java-maven'
```

**Comment by dharmit** [comment]
Looking at the error message, it looks to me like the pod's not getting scheduled since there's no persistent volume to which pod's persistent volume claim can be bound:
```
"reason":"Unschedulable","message":"running \"VolumeBinding\" filter plugin for pod \"java-maven-5fb6944575-mqj8f\": pod has unbound immediate PersistentVolumeClaims"
```

@kadel @mik-dass can you confirm if I'm looking at this correctly?

**Comment by kadel** [comment]
> @kadel @mik-dass can you confirm if I'm looking at this correctly?

yes, it it looks like something is wrong with volumes on the cluster

**Comment by elizabethlumban** [comment]
hi @dharmit - I did follow all the steps listed here, the login and making sure im on the right project and the steps under deploying backend component

https://docs.openshift.com/container-platform/4.7/cli_reference/developer_cli_odo/creating_and_deploying_applications_with_odo/creating-a-multicomponent-application-with-odo.html

**Comment by dharmit** [comment]
@elizabethlumban this is not about just following the steps on the guide you have linked. This looks like an issue about lack of persistent volumes in the cluster you're working on.

* What's the OpenShift cluster setup like? 
  * Is it a minishift or CRC or something deployed in your organization? 
* What's the output of `oc get pvc`?

**Comment by elizabethlumban** [comment]
hi, the ocp cluster is deployed in the organization, I've attached the latest logs and the results of the oc get pv and oc get pvc logs.
[logs01May2021.txt](https://github.com/openshift/odo/files/6408976/logs01May2021.txt)
[pv_pvc.txt](https://github.com/openshift/odo/files/6408977/pv_pvc.txt)

**Comment by dharmit** [comment]
@elizabethlumban the problem is that there are no PVs available to claim in your setup. The `m2-java-maven-uikp` is in `Pending` state since it can't find a Persistent Storage to bind to. All the PVs I see in the `oc get pv` output are already bound to some PVC. You'll have to ask the cluster admin to provide a PV that can be used for your task.
